### PR TITLE
allow PK omitempty

### PR DESCRIPTION
**Pull request by aleksandra-tarkowska**
this PR allow populating maps if one of PK is empty, otherwise entire entry is missing

### CODE REVIEW

**Comment by joshmoore** [comment]
Looking at this, do we want `omitempty` to be the default? cc: @manics

**Comment by aleksandra-tarkowska** [comment]
I decided not because that only affects multiple PK. Otherwise we will end up with more one row empty maps.

File: idr0012-fuchs-cellmorph/screenA/idr0012-screenA-bulkmap-config.yml [hunk 1]
```diff
@@ -47,16 +47,18 @@ columns:
   - group:
       namespace: openmicroscopy.org/mapr/sirna
       columns:
-      - name: siRNA Pool Identifier
-        clientname: siRNA Identifier
+      - name: siRNA Identifier
+        include: yes
+        omitempty: no
+      - name: siRNA Identifier
```
Comment (eleanorwilliams) [inline idr0012-fuchs-cellmorph/screenA/idr0012-screenA-bulkmap-config.yml]: This should be siRNA Pool Identifier. This screen has both columns in the csv

File: idr0012-fuchs-cellmorph/screenB/idr0012-screenB-bulkmap-config.yml [hunk 2]
```diff
@@ -63,12 +63,17 @@ columns:
...
-      - name: siRNA Identifier
+        omitempty: no
+      - name: siRNA Pool Identifier
         include: yes
+        omitempty: no
 
   - group:
       namespace: openmicroscopy.org/mapr/sirna/supplementary
       columns:
+      - name: siRNA Identifier
```
Comment (eleanorwilliams) [inline idr0012-fuchs-cellmorph/screenB/idr0012-screenB-bulkmap-config.yml]: I think we don't need this bit now

File: idr0012-fuchs-cellmorph/screenC/idr0012-screenC-bulkmap-config.yml [hunk 3]
```diff
@@ -61,12 +61,17 @@ columns:
...
-      - name: siRNA Identifier
+        omitempty: no
+      - name: siRNA Pool Identifier
         include: yes
+        omitempty: no
 
   - group:
       namespace: openmicroscopy.org/mapr/sirna/supplementary
       columns:
+      - name: siRNA Identifier
```
Comment (eleanorwilliams) [inline idr0012-fuchs-cellmorph/screenC/idr0012-screenC-bulkmap-config.yml]: We don't need this bit now

File: idr0012-fuchs-cellmorph/screenB/idr0012-screenB-bulkmap-config.yml [hunk 4]
```diff
@@ -63,12 +63,17 @@ columns:
       columns:
       - name: siRNA Pool Identifier
```
Comment (eleanorwilliams) [inline idr0012-fuchs-cellmorph/screenB/idr0012-screenB-bulkmap-config.yml]: This should be siRNA Identifier. This screen has both columns in the csv

File: idr0012-fuchs-cellmorph/screenC/idr0012-screenC-bulkmap-config.yml [hunk 5]
```diff
@@ -61,12 +61,17 @@ columns:
       columns:
       - name: siRNA Pool Identifier
```
Comment (eleanorwilliams) [inline idr0012-fuchs-cellmorph/screenC/idr0012-screenC-bulkmap-config.yml]: This should be siRNA Identifier. This screen has both columns in the csv

File: idr0020-barr-chtog/screenA/idr0020-screenA-bulkmap-config.yml [hunk 6]
```diff
@@ -58,8 +58,11 @@ columns:
       namespace: openmicroscopy.org/mapr/sirna
       columns:
       - name: siRNA Pool Identifier
```
Comment (eleanorwilliams) [inline idr0020-barr-chtog/screenA/idr0020-screenA-bulkmap-config.yml]: this needs the client name of siRNA Identifier

File: idr0008-rohn-actinome/screenB/idr0008-screenB-bulkmap-config.yml [hunk 7]
```diff
@@ -58,6 +58,12 @@ columns:
       columns:
       - name: siRNA Identifier
         include: yes
+        omitempty: no
+      - name: siRNA Identifier
```
Comment (aleksandra-tarkowska) [inline idr0008-rohn-actinome/screenB/idr0008-screenB-bulkmap-config.yml]: that is what simon's example shown. We have to pick up existing header and just put empty value

Comment (eleanorwilliams) [inline idr0008-rohn-actinome/screenB/idr0008-screenB-bulkmap-config.yml]: yes, I agree that is the case when there is only one of the columns in the csv, but in this screen both siRNA Identifier and siRNA Pool Identifier are in the csv.

Comment (aleksandra-tarkowska) [inline idr0008-rohn-actinome/screenB/idr0008-screenB-bulkmap-config.yml]: sorry I looked at 8A :)

**Comment by joshmoore** [comment]
All annotations in on demo3.1

**Pull request merged**